src/de/mangatube/src/eu/kanade/tachiyomi/extension/de/mangatube/MangaTube.kt

```diff
@@ -40,146 +35,225 @@ class MangaTube : ParsedHttpSource() {
-                status = info.select("li:contains(Offiziel)").firstOrNull()?.ownText().toStatus()
-                genre = info.select(".genre-list a").joinToString { it.text() }
-                thumbnail_url = info.select("img").attr("abs:data-original")
-            }
-            description = document.select("div.series-footer h4 ~ p").joinToString("\n\n") { it.text() }
+            title = manga.title
+            author = manga.author.joinToString(", ") { author -> author.name }
+            url = "$baseUrl${manga.url}"
+            artist = manga.artist.joinToString(", ") { artist -> artist.name }
+            description = manga.description
+            thumbnail_url = manga.cover
+            genre = manga.genre.map { genre -> Genre.fromId(genre)!! }
+                .joinToString(", ") { genre -> genre.displayName }
+            status = MangaTubeHelper.mangaStatus(manga.status)
         }
     }
 
-    private fun String?.toStatus() = when {
-        this == null -> SManga.UNKNOWN
-        this.contains("laufend", ignoreCase = true) -> SManga.ONGOING
-        this.contains("abgeschlossen", ignoreCase = true) -> SManga.COMPLETED
-        else -> SManga.UNKNOWN
+    override fun chapterListRequest(manga: SManga): Request {
+        val url = "$baseUrl/api/manga/${mangas[manga.title]!!.slug}/chapters"
+
+        Log.d("MangaTube", "Request chapters for manga: ${manga.title}")
+        Log.d("MangaTube", "Url -> $url")
+
+        return GET(url)
     }
 
-    // Chapters
+    @SuppressLint("SimpleDateFormat")
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val body = MangaTubeHelper.checkResponse(response)
 
-    override fun chapterListSelector() = "ul.chapter-list li"
+        val res: BaseResponse<ChaptersWrapper> = json.decodeFromString(body)
 
-    override fun chapterFromElement(element: Element): SChapter {
-        return SChapter.create().apply {
-            element.select("a[title]").let {
-                name = "${it.select("b").text()} ${it.select("span:not(.btn)").joinToString(" ") { span -> span.text() }}"
-                setUrlWithoutDomain(it.attr("href"))
-            }
-            date_upload = element.select("p.chapter-date").text().let {
-                try {
-                    SimpleDateFormat("dd.MM.yyyy", Locale.getDefault()).parse(it.substringAfter(" "))?.time ?: 0L
-                } catch (_: ParseException) {
-                    0L
-                }
+        if(!res.success){
+            throw Exception("Something went wrong!")
+        }
+
+        val chapterList = res.data.chapters.map { chapter ->
+            SChapter.create().apply {
+                url = "$baseUrl${chapter.readerURL}"
+                name = chapter.name.ifBlank { "Chapter ${chapter.number}" }
+                date_upload =
+                    SimpleDateFormat("yyyy-MM-dd HH:mm:ss").parse(chapter.publishedAt)!!.time
```

Comment:
initialize `SimpleDateFormat in the class to avoide initializing it every time, and wrap the parsing in try catch to avoide breaking in case of bad dates